### PR TITLE
Update healthcare logic

### PR DESCRIPTION
**Pull request by JoshElgar**
Mainly updates the HealthWorkerExposureScreen and the About Your Work screens to add a new question (In the last day have you interacted with any patients?), switch two choices for the healthcareStaffOptions and accordingly change the validation for following fields. 

Small changes:
Removed isCompact from dropdown field as it was serving no purpose aside from breaking styling. There is already an onlyPicker prop that can be used if you only need the picker component and not the styling around it.

Added some labels / required validation messages.

### CODE REVIEW

File: src/core/user/UserService.ts [hunk 1]
```diff
@@ -204,7 +204,7 @@ export default class UserService extends ApiClientBase {
             return false
         } else {
             await AsyncStorageService.setIsHealthWorker(
-                (patientProfileResponse.data.healthcare_professional === "yes_does_treat")
+                (patientProfileResponse.data.healthcare_professional === "yes_does_interact")
```
Comment (abubuwe) [inline src/core/user/UserService.ts:207]: we should keep the old option - some users will log-in with that value returned and should still see the screen,

File: src/features/assessment/HealthWorkerExposureScreen.tsx [hunk 2]
```diff
@@ -67,46 +69,38 @@ export default class HealthWorkerExposureScreen extends Component<HealthWorkerEx
...
+            then: Yup.string().required(i18n.t("required-has-used-ppe-equipment")),
+        }),
+        ppeAvailabilityAlways: Yup.string().when(['interactedAnyPatients', 'hasUsedPPEEquipment'], {
+            is: (interactedAnyPatients, hasUsedPPEEquipment) => interactedAnyPatients === 'yes' && hasUsedPPEEquipment === 'always',
+            then: Yup.string().required(i18n.t("required-ppe-availability-always")),
         }),
         ppeAvailabilitySometimes: Yup.string().when('hasUsedPPEEquipment', {
-            is: 'sometimes',
-            then: Yup.string().required()
+            is: (interactedAnyPatients, hasUsedPPEEquipment) => interactedAnyPatients === 'yes' && hasUsedPPEEquipment === 'sometimes',
```
Comment (abubuwe) [inline src/features/assessment/HealthWorkerExposureScreen.tsx:98]: you expect two params here but your when only contains 1 form value?

File: src/features/assessment/HealthWorkerExposureScreen.tsx [hunk 3]
```diff
@@ -67,46 +69,38 @@ export default class HealthWorkerExposureScreen extends Component<HealthWorkerEx
...
+            then: Yup.string().required(i18n.t("required-ppe-availability-sometimes"))
         }),
         ppeAvailabilityNever: Yup.string().when('hasUsedPPEEquipment', {
-            is: 'never',
-            then: Yup.string().required()
+            is: (interactedAnyPatients, hasUsedPPEEquipment) => interactedAnyPatients === 'yes' && hasUsedPPEEquipment === 'never',
```
Comment (abubuwe) [inline src/features/assessment/HealthWorkerExposureScreen.tsx:102]: you expect two params here but your when only contains 1 form value?

File: src/core/user/UserService.ts (hunk 1 above)
Comment (JoshElgar) [reply src/core/user/UserService.ts:207]: Good point

**Pull request merged**